test/envtest/k8s_objects_status_envtest_test.go

```diff
@@ -127,12 +135,20 @@ func runManagerWithConfig(ctx context.Context, t *testing.T, envcfg *rest.Config
 	cfg.FeatureGates[featuregates.GatewayFeature] = true
 	cfg.FeatureGates[featuregates.GatewayAlphaFeature] = true
 
-	modifyCfgFn(&cfg)
+	if modifyCfgFn != nil {
+		modifyCfgFn(&cfg)
+	}
```

Comment:
How about instead of this make this function accept option funcs which can modify the `*manager.Config`?

This way there's no awkward `nil` in the argument list when you don't want to modify it. In the long run we'll be able to make this a bit more formalized and reuse in other places as well. WDYT?